services/outputhost/ackmanager.go

```diff
@@ -65,12 +59,11 @@ type (
 		cgUUID             string
 		extUUID            string
 		connectedStoreUUID *string
-		*levels                    // ‡ the current levels
-		prev               *levels // ‡ the previous levels
+		*levels            // ‡ the current levels
 		ackLevelTicker     *time.Ticker
 		closeChannel       chan struct{}
 		waitConsumed       chan<- bool // waitConsumed is the channel which will signal if the extent is completely consumed given by extentCache
-		metaclient         metadata.TChanMetadataService
+		committer          Committer
```

Comment:
hmmm.. ok.. may be we can iterate on it in the next pass.. I was thinking more of something like this but this will of course need lots of rearrangements of the code:
`Committer {
    ackMgr *StructCommonAckManager // this is the common ack manager struct between kafka and cherami but lives within the committer with locks being controlled by the committer..
    lk            sync.RWMutex
   <other necessary fields>
}`